rules/go_rules.build_defs

```diff
@@ -1667,3 +1694,45 @@ def _get_ldflags_and_pkgconfig(name):
     pkg_config = ' '.join([l[3:] for l in labels if l.startswith('pc:')])
     return (ldflags, f'`pkg-config --libs {pkg_config}`') if pkg_config else (ldflags, '')
 
+
+def _debug_cmd(name:str, bin:str, flags:str='', pre_cmd:str='', srcs:list=[], deps:list=[], test_only:bool=False):
+    if not CONFIG.get("DEBUG"):
+        return "", None, None
+
+    cmd = f"$DEBUG_TOOLS_DLV exec {bin}"
+    data = []
+    tools = {
+        'dlv': CONFIG.DELVE_TOOL
+    }
+
+    if pre_cmd:
+        cmd = f"{pre_cmd} && {cmd}"
+
+    port = CONFIG.DEBUG.PORT
+    if not port:
+        # A CLI debug session needs access to the source files for improved UX.
+        debug_srcs_out = f"_{name}_debug_srcs"
+        data = [
+            build_rule(
+                name = name,
+                tag = "debug_srcs",
+                srcs = srcs,
+                # Grab all sources into a directory.
+                cmd = "find . -name \"*.go\" | cut -c 3- | xargs -I {} sh -c 'mkdir -p $OUTS/$(dirname {}) && cp {} $OUTS/$(dirname {})'",
+                outs = [debug_srcs_out],
+                deps = deps,
+                test_only = test_only,
+                requires = ["go_src"],
+                needs_transitive_deps = True,
+            )
+        ]
+        # Extract sources before running the debugger.
+        cmd = f"cp -R $PKG_DIR/{debug_srcs_out}/* . && {cmd}"
```

Comment:
This can get slow for targets with lots of transitive dependencies. Not a big problem for debugging, but an area for improvement.

Reply:
Replaced this with packing and unpacking of sources, yielding much faster results.